apps/sms/services/js/threads.js

```diff
@@ -6,16 +6,24 @@
   var threads = new Map();
   var messageMap = new Map();
 
-  function Thread(thread) {
-    var length = Thread.FIELDS.length;
-    var key;
-
-    for (var i = 0; i < length; i++) {
-      key = Thread.FIELDS[i];
-      this[key] = thread[key];
-    }
+  function extractSupportedFields(
+    { id, body, timestamp, participants, lastMessageType, status, isDraft }
+  ) {
+    return {
+      id,
+      body,
+      timestamp,
+      participants,
+      lastMessageType,
+      status,
+      isDraft: !!isDraft
```

Comment:
I think it's good as a "top level" property; this is really one of the main characterstics of the object